index.js

```diff
@@ -107,53 +107,61 @@ class SimpleKeyring extends EventEmitter {
   }
 
   // personal_signTypedData, signs data along with the schema
-  signTypedData_v1 (withAccount, typedData, opts = {}) {
-    const privKey = this.getPrivateKeyFor(withAccount, opts);
+  signTypedData_v1 (withAccount, typedData) {
+    const privKey = this.getPrivateKeyFor(withAccount);
     const sig = sigUtil.signTypedDataLegacy(privKey, { data: typedData })
     return Promise.resolve(sig)
   }
 
   // personal_signTypedData, signs data along with the schema
-  signTypedData_v3 (withAccount, typedData, opts = {}) {
-    const privKey = this.getPrivateKeyFor(withAccount, opts);
+  signTypedData_v3 (withAccount, typedData) {
+    const privKey = this.getPrivateKeyFor(withAccount);
     const sig = sigUtil.signTypedData(privKey, { data: typedData })
     return Promise.resolve(sig)
   }
 
   // personal_signTypedData, signs data along with the schema
-  signTypedData_v4 (withAccount, typedData, opts = {}) {
-    const privKey = this.getPrivateKeyFor(withAccount, opts);
+  signTypedData_v4 (withAccount, typedData) {
+    const privKey = this.getPrivateKeyFor(withAccount);
     const sig = sigUtil.signTypedData_v4(privKey, { data: typedData })
     return Promise.resolve(sig)
   }
 
-  getPrivateKeyFor (address, opts = {}) {
+  getPrivateKeyFor (address) {
     if (!address) {
       throw new Error('Must specify address.');
     }
-    const wallet = this._getWalletForAccount(address, opts)
+    const wallet = this._getWalletForAccount(address)
     const privKey = ethUtil.toBuffer(wallet.getPrivateKey())
     return privKey;
   }
 
-  // returns an address specific to an app
-  getAppKeyAddress (address, origin) {
-    return new Promise((resolve, reject) => {
-      try {
-        const wallet = this._getWalletForAccount(address, {
-          withAppKeyOrigin: origin,
-        })
-        const appKeyAddress = sigUtil.normalize(wallet.getAddress().toString('hex'))
-        return resolve(appKeyAddress)
-      } catch (e) {
-        return reject(e)
-      }
-    })
+  async getAppKey (address, origin) {
+    if (
+      !origin ||
+      typeof origin !== 'string'
+    ) {
+      throw new Error(`'origin' must be a non-empty string`)
+    }
+
+    const wallet = this._getWalletForAccount(address)
+    const privKey = wallet.getPrivateKey()
+    const appKeyOriginBuffer = Buffer.from(origin, 'utf8')
+    const appKeyBuffer = Buffer.concat([privKey, appKeyOriginBuffer])
+    const appKeyPrivKey = ethUtil.keccak(appKeyBuffer, 256)
+    return appKeyPrivKey
```

Comment:
This currently returns a `Buffer`. Let me know if it'd be more appropriate to return a hex string instead, like `exportAccount` does.

Reply:
don't know the specifics, but as a general rule, keep new functionality aligned with existent one.

Reply:
Yeah I'd love if we used buffers internally, but until we do, I lean towards being more consistent in the interface.